### PR TITLE
tests: use mock FutureScheduler in tests instead of Option<FutureSche…

### PR DESCRIPTION
**Pull request by friedm**
New to the project and making an attempt at #2886.

Avoids using an `Option<FutureScheduler>` in `Storage::ReadPoolContext`, which was done only for testing purposes. When tests create a read pool, they use a test `FutureScheduler` instead of `None`.

### CODE REVIEW

**Comment by siddontang** [comment]
Very cool, thanks @friedm 

PTAL @breeswish

**Comment by breeswish** [comment]
Thanks for your contribution! I suggest to use `let xxx = FutureWorker::new(..);` and then `xxx.scheduler()` to get a mock scheduler, like what we did in the endpoint.

File: src/util/worker/future.rs [hunk 1]
```diff
@@ -55,7 +55,7 @@ pub struct Scheduler<T> {
 }
 
 impl<T: Display> Scheduler<T> {
-    fn new<S: Into<String>>(name: S, sender: UnboundedSender<Option<T>>) -> Scheduler<T> {
+    pub fn new<S: Into<String>>(name: S, sender: UnboundedSender<Option<T>>) -> Scheduler<T> {
```
Comment (breeswish) [inline src/util/worker/future.rs]: You can revert this change now since we are not calling `Scheduler::new` any more. Rest LGTM.

**Comment by breeswish** [comment]
LGTM
